apiserver/facades/agent/upgradeseries/upgradeseries.go

```diff
@@ -121,6 +121,32 @@ func (a *API) SetMachineStatus(args params.UpgradeSeriesStatusParams) (params.Er
 	return result, nil
 }
 
+// CurrentSeries returns what Juju thinks the current series of the machine is.
+// Note that a machine could have been upgraded out-of-band by running
+// do-release-upgrade outside of the upgrade-series workflow,
+// making this value incorrect.
+func (a *API) CurrentSeries(args params.Entities) (params.StringResults, error) {
+	result := params.StringResults{}
+
+	canAccess, err := a.AccessMachine()
+	if err != nil {
+		return result, err
+	}
+
+	results := make([]params.StringResult, len(args.Entities))
+	for i, entity := range args.Entities {
+		machine, err := a.authAndMachine(entity, canAccess)
```

Comment:
small nit: why not `getMachine`? authAndMachine is a rather odd name for a method ;-)

Reply:
Settled on `authAndGetMachine`.